### PR TITLE
Implement DCF modal for search and other nav toggle cleanup

### PR DESCRIPTION
**Pull request by jsturek**

### CODE REVIEW

**Comment by jsturek** [comment]
TODOs
* Finalize search markup changes
* Fix Axe and W3C test failures
* Verify all dead toggle and modal has been removed

File: wdn/templates_5.0/includes/global/search.html [hunk 1]
```diff
@@ -1,18 +1,20 @@
...
   </div>
-  <div class="unl-search-modal dcf-fixed dcf-pin-top dcf-pin-left dcf-h-100% dcf-w-100% dcf-bg-overlay-light" id="dcf-search-results" aria-labelledby="unl-search-modal-heading" aria-hidden="true" role="dialog" tabindex="-1" hidden>
-    <div class="dcf-d-flex dcf-flex-col dcf-flex-nowrap dcf-h-100% dcf-w-100%" role="document">
-      <h2 class="dcf-sr-only" id="unl-search-modal-heading">Search Form</h2>
-      <div class="dcf-bleed dcf-bg-center dcf-bg-no-repeat dcf-bg-cover unl-bg-scarlet unl-search-bg">
+  <div id="dcf-search-results" class="dcf-modal dcf-bg-overlay-light dcf-fixed dcf-pin-top dcf-pin-left dcf-h-100% dcf-w-100% dcf-d-flex dcf-ai-center dcf-jc-center dcf-opacity-100 dcf-pointer-events-auto" hidden>
+    <div class="dcf-modal-wrapper dcf-h-max-100vh dcf-relative dcf-h-auto dcf-overflow-y-auto dcf-d-flex dcf-flex-col dcf-flex-nowrap dcf-h-100% dcf-w-100%">
+      <header class="dcf-modal-header dcf-bleed dcf-bg-center dcf-bg-no-repeat dcf-bg-cover unl-bg-scarlet unl-search-bg">
```
Comment (skoolbus39) [inline wdn/templates_5.0/includes/global/search.html]: Changing this from a `<header>` to a `<div>` should fix the Axe errors and still work with the DCF modal.

Comment (jsturek) [reply wdn/templates_5.0/includes/global/search.html]: done.

**Comment by jsturek** [comment]
@skoolbus39 - This is ready for your review.  Please verify the markup is correct and that all of the modals are behaving correctly in both display and accessibility markup states.

File: wdn/templates_5.0/js-src/search.babel.js [hunk 2]
```diff
@@ -23,64 +21,46 @@ define(['wdn', 'require', 'plugins/body-scroll-lock'], function(WDN, require, bo
...
+        }
       }
 
-      let domDialog = document.getElementById('dcf-search-results');
-      if (!domDialog) {
-        domDialog = document.createElement('div');
-        domDialog.setAttribute('id', 'dcf-search-results');
+      if (domMobileSearchLink && domMobileSearchBtns && domMobileSearchBtns.length) {
+        domMobileSearchLink.setAttribute('hidden', '');
+        domMobileSearchLink.setAttribute('aria-hidden', true);
```
Comment (skoolbus39) [inline wdn/templates_5.0/js-src/search.babel.js]: `aria-hidden="true"` isn't necessary. The `hidden` attribute should suffice.

File: wdn/templates_5.0/js-src/navigation.babel.js [hunk 3]
```diff
@@ -83,11 +81,14 @@ define(['plugins/headroom', 'plugins/body-scroll-lock'], function(Headroom, body
...
+        document.dispatchEvent(closeSearchEvent);
+        document.dispatchEvent(closeIDMOptionsEvent);
+
         if (window.matchMedia("(max-width: 56.12em)").matches) {
           skipNav.setAttribute('aria-hidden', 'true');
           institutionTitle.setAttribute('aria-hidden', 'true');
-          idm.setAttribute('aria-hidden', 'true');
-          search.setAttribute('aria-hidden', 'true');
           logo.setAttribute('aria-hidden', 'true');
           nav.setAttribute('aria-hidden', 'true');
```
Comment (skoolbus39) [inline wdn/templates_5.0/js-src/navigation.babel.js]: The navigation should be set to `aria-hidden="false"` when open.

File: wdn/templates_5.0/js-src/navigation.babel.js [hunk 4]
```diff
@@ -115,12 +112,13 @@ define(['plugins/headroom', 'plugins/body-scroll-lock'], function(Headroom, body
         if (window.matchMedia("(max-width: 56.12em)").matches) {
           skipNav.setAttribute('aria-hidden', 'false');
           institutionTitle.setAttribute('aria-hidden', 'false');
-          idm.setAttribute('aria-hidden', 'false');
-          search.setAttribute('aria-hidden', 'false');
           logo.setAttribute('aria-hidden', 'false');
           nav.setAttribute('aria-hidden', 'false');
```
Comment (skoolbus39) [inline wdn/templates_5.0/js-src/navigation.babel.js]: The navigation should be set to `aria-hidden="true"` when closed – _and_ by default when the window is `max-width: 56.12em`.

**Comment by jsturek** [comment]
@skoolbus39, requested changes made. Ready for another look.

**Comment by jsturek** [comment]
@skoolbus39, ready for another look.  I updated to use the modal functionality to update the nav toggle button state.

**Pull request merged**